### PR TITLE
ref(ui): Use trailingItems prop in team selector

### PR DESCRIPTION
**Pull request by vuluongj20**
Move `<AddToProjectButton />` from the `label` option prop to the `trailingItems` prop, and vertically align it.

**Before:**
Screen Shot 2022-01-12 at 1 35 14 PM

**After:**
Screen Shot 2022-01-12 at 1 35 40 PM

### CODE REVIEW

File: static/app/components/forms/selectControl.tsx [hunk 1]
```diff
@@ -81,6 +81,12 @@ export type ControlProps<OptionType = GeneralSelectValue> = Omit<
    * Show line dividers between options
    */
   showDividers?: boolean;
+  /**
+   * If false (default), checkmarks/checkboxes will be vertically centered
+   * wrt the first line of the label text. If true, they will be centered
+   * wrt the entire height of the option wrap.
+   */
+  verticallyCenterCheckWrap?: boolean;
```
Comment (vuluongj20) [inline static/app/components/forms/selectControl.tsx:89]: Useful for when we have leading items that are taller than the label line height.

When `false`:

Screen Shot 2022-01-12 at 1 37 50 PM

When `true`:

Screen Shot 2022-01-12 at 1 37 22 PM

File: static/app/components/forms/selectOption.tsx [hunk 2]
```diff
@@ -3,14 +3,20 @@ import styled from '@emotion/styled';
 
 import {IconCheckmark} from 'sentry/icons';
 import space from 'sentry/styles/space';
+import {defined} from 'sentry/utils';
 
 type Props = React.ComponentProps<typeof selectComponents.Option>;
 
 function SelectOption(props: Props) {
+  const {data} = props;
```
Comment (davidenwang) [inline static/app/components/forms/selectOption.tsx:11]: is there a reason we don't destructure more props in here like isFocus, isMulti?

**Review by davidenwang — APPROVED** [review]
looking good!

File: static/app/components/forms/selectOption.tsx (hunk 2 above)
Comment (vuluongj20) [reply static/app/components/forms/selectOption.tsx]: good point, just did!

**Pull request merged**